core/modules/replica/Controller.h

```diff
@@ -0,0 +1,687 @@
+     * @param allowDuplicate   - follow a previously made request if the current one duplicates it
+     * @param jobId            - an optional identifier of a job issed the request
+     * @param requestExpirationIvalSec
+     *                         - an optional parameter (if differs from 0)
+     *                           allowing to override the default value of the corresponding
+     *                           parameter from the Configuration.
+     *
+     * @return a pointer to the replication request
+     */
+    ReplicationRequest_pointer replicate(std::string const& workerName,
+                                         std::string const& sourceWorkerName,
+                                         std::string const& database,
+                                         unsigned int chunk,
+                                         ReplicationRequest_callback_type onFinish=nullptr,
+                                         int  priority=0,
+                                         bool keepTracking=true,
+                                         bool allowDuplicate=true,
+                                         std::string const& jobId="",
+                                         unsigned int requestExpirationIvalSec=0);
+
+    /**
+     * Initiate a new replica deletion request.
+     *
+     * @param workerName     - the name of a worker node where the replica will be deleted
+     * @param database       - database name
+     * @param chunk          - the chunk number
+     * @param onFinish       - an optional callback function to be called upon the completion of the request
+     * @param priority       - a priority level of the request
+     * @param keepTracking   - keep tracking the request before it finishes or fails
+     * @param allowDuplicate - follow a previously made request if the current one duplicates it
+     * @param jobId          - an optional identifier of a job issed the request
+     * @param requestExpirationIvalSec
+     *                       - an optional parameter (if differs from 0)
+     *                         allowing to override the default value of the corresponding
+     *                         parameter from the Configuration.
+     *
+     * @return a pointer to the replication request
+     */
+    DeleteRequest_pointer deleteReplica(std::string const& workerName,
+                                        std::string const& database,
+                                        unsigned int chunk,
+                                        DeleteRequest_callback_type onFinish=nullptr,
+                                        int  priority=0,
+                                        bool keepTracking=true,
+                                        bool allowDuplicate=true,
+                                        std::string const& jobId="",
+                                        unsigned int requestExpirationIvalSec=0);
+
+    /**
+     * Initiate a new replica lookup request.
```

Comment:
I'm experiencing some cognitive dissonance between "lookup" in the comments and "find" in the function name. I'd prefer same word was used both places.

Reply:
Fixed